### PR TITLE
nostui: init 0.1.0

### PR DESCRIPTION
**Pull request by heywoodlh**
## Description of changes

Adds nostui package

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by heywoodlh** [comment]
Output of `nixpkgs-review pr 328039`

```
1 package added:
nostui (init at 0.1.0)
...
1 package built:
nostui
```

File: pkgs/by-name/no/nostui/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+, stdenv
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "nostui";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "akiomik";
+    repo = pname;
```
Comment (pbsds) [inline pkgs/by-name/no/nostui/package.nix:14]: this is considered an antipattern

```suggestion
    repo = "nostui";
```

File: pkgs/by-name/no/nostui/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-RCD11KdzM66Mkydc51r6fG+q8bmKl5eZma58YoARwPo=";
```
Comment (pbsds) [inline pkgs/by-name/no/nostui/package.nix:16]:
```suggestion
    hash = "sha256-RCD11KdzM66Mkydc51r6fG+q8bmKl5eZma58YoARwPo=";
```

File: pkgs/by-name/no/nostui/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "--skip=widgets::text_note::tests::test_created_at"
+  ];
+
+  cargoHash = "sha256-8ciA1FoGdnU+GWRcYJc8zU2FpUgGwTZSZynvAi1luYo=";
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  meta = with lib; {
```
Comment (pbsds) [inline pkgs/by-name/no/nostui/package.nix:36]: please set `mainProgram`

File: pkgs/by-name/no/nostui/package.nix (hunk 1 above)
Comment (heywoodlh) [reply pkgs/by-name/no/nostui/package.nix]: Fixed in latest push, thanks for the feedback!

File: pkgs/by-name/no/nostui/package.nix (hunk 2 above)
Comment (heywoodlh) [reply pkgs/by-name/no/nostui/package.nix]: Fixed in latest push.

File: pkgs/by-name/no/nostui/package.nix (hunk 3 above)
Comment (heywoodlh) [reply pkgs/by-name/no/nostui/package.nix:36]: Fixed in latest push.

**Review by pbsds — APPROVED** [review]

**Pull request merged**